src/database/alert_dao.rs

```diff
@@ -550,4 +593,163 @@ mod tests {
         assert_eq!(loaded_alerts.len(), 1);
         assert_eq!(loaded_alerts[0], alert1);
     }
+
+    #[test]
+    fn test_marks_alert_as_read() {
+        let database = Arc::new(Database::new(
+            Connection::open_in_memory().expect("Couldn't create database!"),
+        ));
+        let alert_dao = AlertDaoImpl::new(database.clone());
```

Comment:
immutable api preferred, right?

Reply:
In Rust not necessarily, as mutability is safer than Swift/Kotlin.

Not sure how it relates with this code though?

Reply:
@i-schuetz 
> In Rust not necessarily, as mutability is safer than Swift/Kotlin.
> 
> Not sure how it relates with this code though?

I was wondering why you prefer passing _database.clone()_ as argument, as opposed to just _database_ . Is it due adopted project convention, or ...?

Reply:
Ah, no, in this case it's probably a copy-paste leftover from the database injection in the composition root. Cloning is not needed here as it's only used by `AlertDaoImpl`

Reply:
https://github.com/Co-Epi/app-backend-rust/pull/143